components/client/src/test/unit/java/com/hotels/styx/client/loadbalancing/strategies/PowerOfTwoStrategyTest.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (C) 2013-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.styx.client.loadbalancing.strategies;
+
+import com.google.common.collect.Iterables;
+import com.hotels.styx.api.HttpClient;
+import com.hotels.styx.api.client.ActiveOrigins;
+import com.hotels.styx.api.client.ConnectionPool;
+import com.hotels.styx.api.client.Origin;
+import com.hotels.styx.api.client.RemoteHost;
+import com.hotels.styx.api.client.loadbalancing.spi.LoadBalancingStrategy;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+
+import static com.google.common.collect.Iterables.getFirst;
+import static com.hotels.styx.api.client.Origin.newOriginBuilder;
+import static com.hotels.styx.api.client.RemoteHost.remoteHost;
+import static com.hotels.styx.api.support.HostAndPorts.localHostAndFreePort;
+import static java.util.Arrays.asList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+
+public class PowerOfTwoStrategyTest {
+    final Origin ORIGIN_ONE = newOriginBuilder(localHostAndFreePort()).id("one").build();
```

Comment:
1. These field names are capitalised, which indicates static fields in Java, but these fields are not static.
2. We should have consistent visibility between our constants. Here, some are private, some are default.